examples/nsl_pk/DY.Example.NSL.Protocol.Stateful.fst

```diff
@@ -60,7 +68,7 @@ type nsl_global_sess_ids = {
 
 val prepare_msg1: principal -> principal -> traceful state_id
 let prepare_msg1 alice bob =
-  let* n_a = mk_rand NoUsage (join (principal_label alice) (principal_label bob)) 32 in
+  let* n_a = mk_rand NoUsage (join (nsl_nonce_label alice) (nsl_nonce_label bob)) 32 in
```

Comment:
Somehow, the name `nsl_nonce_label` is confusing to me.
From the name, I would expect that it stands for the label of the nonces used in NSL. However, the label of the nonces (as can be seen here for n_a) are `join alice bob`.
Why is the "renaming" of `principal_label` to `nsl_nonce_label` helpful?

Reply:
Correct, that's not a great name, `nsl_nonce_label` is actually only "half" of the nonce label. I gave it a name because I think it's best practice giving a name to labels, but we can either choose to not care about that for a simple example and go back to `principal_label`, or add the `join` in `nsl_nonce_label alice bob`: it seems that `nsl_nonce_label` is always used with a `join` so that would work.

Reply:
I think, having the `join` inside the `nsl_nonce_label` would be good

Reply:
Done